### DISCUSSION TITLE
blog/2024-02-10_keycloak-docker-compose-nginx/

### DISCUSSION THREAD

**Discussion by giscus** [Sieboldianus/du.nkel.dev#17] [Announcements]
# blog/2024-02-10_keycloak-docker-compose-nginx/

A personal code notes blog

https://du.nkel.dev/blog/2024-02-10_keycloak-docker-compose-nginx/

**Comment by Sieboldianus** [comment]
Comments welcome. If you do not want to allow Github API access for [giscus.app](https://giscus.app/), feel free to [directly add comments on Github](https://github.com/Sieboldianus/du.nkel.dev/discussions/17).

**Comment by euandmj** [comment]
I would highly recommend against Keycloak. It's user experience is iffy at best, it's API is misdocumented and has atypical behaviour, amongst other problems. But more importantantly there are known security issues and an exponentially scaling performance issue with multi tenancy that renders the API and web portal unusable.

Best to consider alternatives but great article none the less

**Reply by EmilPasca** [reply]
I've been using Keycloak to learn more about OAuth, PKCE and OpenID. I've read about some of the security issues, Keycloak has, but I don't know any other open source tool with so many features like this one. What would you suggest as an alternative?

**Reply by Sieboldianus** [reply]
I agree to @EmilPasca. There was a [comment](https://news.ycombinator.com/item?id=39340691) on HN regarding a similar concern for some of the CVEs found in Keycloak:

> It's a double edged sword. A closed source non-transparent solution without a reported cve is also laughable as "secure".

Everyone has to consider these trade-offs, there is no one-fits-all solution available. I think having an open source software that is widely used is preferable, from a security perspective, over any less used or closed source solution. I found Keycloak pretty robust, easy to setup and covering most of my use cases, which is why I selected it. YMMV.

**Comment by generalistcodes** [comment]
If you experience issues when attempting to connect to an external database (such as RDS), removing the `--optimized` flag may resolve these problems.
